plugins/reach/lib/kReachManager.php

```diff
@@ -46,55 +46,63 @@ protected function getObjectType($eventObjectClassName)
 
 	private function addingEntryVendorTaskByObjectIds($taskObjectType,  $allowedCatalogItemIds, $profileId, $object)
 	{
-		$vendorTaskObjectHandler = HandlerFactory::getHandlerAutomaticFlow($taskObjectType, $object);
+		$vendorTaskObjectHandler = HandlerFactory::getHandler($taskObjectType);
 		if(!$vendorTaskObjectHandler)
 		{
 			KalturaLog::log("Could not get vendor task handler for type {$taskObjectType}");
 			return true;
 		}
-		$taskObjectId = $vendorTaskObjectHandler->getTaskObjectId($object);
+		$taskObjects = $vendorTaskObjectHandler->getTaskObjectsByEventObject($object);
 		$catalogItemIdsToAdd = array_unique($allowedCatalogItemIds);
-		$taskObject = $vendorTaskObjectHandler->retrieveObject($taskObjectId);
 		$reachProfile = ReachProfilePeer::retrieveActiveByPk($profileId);
-		//If both the entry and reach profile don't exist, there's no need to hit the loop
-		if(!$taskObject || !$reachProfile)
+		//If both the task objects and reach profile don't exist, there's no need to hit the loop
```

Comment:
should be "if any"
can remove the comment all together since the log message is enough